app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if not self.hidden:
+            self.hidden = True
+        else:
+            self.hidden = False
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(other: Animal) -> None:
+        if isinstance(other, Herbivore) and not other.hidden:
+            other.health -= 50
+            if other.health <= 0:
+                Animal.alive.remove(other)
```

Comment:
This check can be done for whole bite method not only for Herbivore (becuase we are awaiting Animal as `other`). But that's a recomedation for future. No need to fix it now. 
reason: in future if you will add something like biting other Carnivores or something like that - you will need to duplicate the same check to the other if statement.

Reply:
Understood. Thank you very much!